moped-editor/src/views/projects/projectView/NoteInput.js

```diff
@@ -0,0 +1,236 @@
+  root: {
+    width: "100%",
+    backgroundColor: theme.palette.background.paper,
+  },
+  cancelButton: {
+    margin: theme.spacing(1),
+    position: "relative",
+    color: theme.palette.secondary.dark,
+  },
+  contentEditable: {
+    height: "300px",
+    width: "100%",
+    padding: "8px 8px",
+    border: "1px solid #fff"
+  },
+  placeholder: {
+    position: "absolute",
+    top: "24px",
+    paddingLeft: "8px"
+  },
+  editorWrapper: {
+    position: "relative"
+  },
+  showButtonItem: {
+    margin: theme.spacing(2),
+  },
+}));
+
+/**
+ * Defines the NoteTypeRadioButtons that allow a user to select the note type of a
+ * note they are creating or editing.
+ * @param {integer} defaultValue - The note type id that determines which button option
+ * should be selected/highlighted by default
+ * @param {function} onChange - Callback function that runs when the note type is changed via button click
+ * @return {JSX.Element}
+ * @constructor
+ */
+const NoteTypeRadioButtons = ({ defaultValue, onChange }) => (
+  <RadioGroup
+    row
+    value={defaultValue}
+    onChange={onChange}
+    sx={{ color: "black" }}
+  >
+    <FormControlLabel value={1} control={<Radio />} label="Internal note" />
+    <FormControlLabel value={2} control={<Radio />} label="Status update" />
+  </RadioGroup>
+);
+
+const theme = {};
+
+// Catch any errors that occur during Lexical updates and log them
+// or throw them as needed.
+const onError = (error) => {
+  console.error(error);
+}
+
+// On change, return editor content as HTML
+const OnChangePlugin = ({ onChange }) => {
+  const [editor] = useLexicalComposerContext();
+  useEffect(() => {
+    return editor.registerUpdateListener(({ editorState }) => {
+      editorState.read(() => {
+        const htmlContent = $generateHtmlFromNodes(editor, null);
+        onChange(htmlContent);
+      });
+    });
+  });
+  return null;
+}
+
+// On successful save, clear and refocus the editor
+const OnSavePlugin = ({ noteAddSuccess }) => {
+  const [editor] = useLexicalComposerContext();
+  useEffect(() => {
+    noteAddSuccess && editor.update(() => {
+      const root = $getRoot();
+      root.clear();
+    });
+    editor.focus();
+  }, [noteAddSuccess, editor]);
+  return null;
+}
+
+// On edit, load the selected comment and refocus the editor
+const OnEditPlugin = ({ htmlContent, editingNote }) => {
+  const [editor] = useLexicalComposerContext();
+  useEffect(() => {
+    const editorState = editor.getEditorState()
+    const nodeMap = editorState._nodeMap;
+    const textArray = [...nodeMap.entries()][2];
+    const isTextInEditor = !!textArray;
+    const isTextinDatabase = !!htmlContent.replace(/<[^>]*>/g, "");
+    // If user clicks edit and there is a stored note and a clear editor,
+    // populate editor with stored note
+    editingNote && !isTextInEditor && isTextinDatabase &&
+      editor.update(() => {
+        $getRoot()
+          .getChildren()
+          .forEach((n) => n.remove());
```

Comment:
might be good to add comments to explain what's going on here 👀

Reply:
yes - added a couple step by step comments to clarify this!